platform/ios/src/MGLMapView.mm

```diff
@@ -914,17 +914,21 @@ - (void)updateConstraints
         // logo view
         [self removeConstraints:self.logoViewConstraints];
         [self.logoViewConstraints removeAllObjects];
-        [self.logoViewConstraints addObject:[safeAreaLayoutGuide.bottomAnchor constraintEqualToAnchor:self.logoView.bottomAnchor
-                                                                                             constant:8.0 + self.contentInset.bottom]];
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wpartial-availability"
+        [NSLayoutConstraint activateConstraints:@[[safeAreaLayoutGuide.bottomAnchor constraintEqualToSystemSpacingBelowAnchor:self.logoView.bottomAnchor multiplier:1.0]]];
```

Comment:
This seems to accomplish the same thing as with `compassViewConstraints` above, but takes a different approach: activating the new constraint immediately. Is one way better/different than the other?

Reply:
Looks like the correct way to add a bottom constraint on iOS 11 is anchor the view to the `safeAreaLayoutGuide` I was doing the opposite and it was not working.

```
[safeAreaLayoutGuide.bottomAnchor constraintEqualToSystemSpacingBelowAnchor:self.logoView.bottomAnchor multiplier:1.0]
```

Reply:
Sounds good. However, while we're in this code, can you refactor it to eliminate the duplication? Now that we have a test in place hopefully it feels less risky to do so.

Reply:
> activating the new constraint immediately

I didn't quite understand - is this necessary @fabian-guerra ?

Reply:
This code is no longer valid. Researching the reasons for why the constraint was not applied I tried to force activating the constraint that's why I added this code. If you take a second look you will see that I no longer use this.